src/poetry/installation/executor.py

```diff
@@ -604,7 +615,13 @@ def _prepare_git_archive(self, operation: Install | Update) -> Path:
 
         package._source_url = original_url
 
-        return archive
+        if operation.package.develop:
+            return archive
+        else:
+            cached_archive = cache_dir / archive.name
+            shutil.copy(archive, cached_archive)
```

Comment:
Can we move instead of copy? Alternatively, we could pass an optional destination to `_prepare_archive`.

Reply:
I ended up using the second suggestions, refactoring artifacts cache a bit, making git cache use the same logic to find correct archive as is done for ordinary packages. This is more safe then just grabbing the first wheel found in the cache directory.